packages/cli/src/eventbus/audit-event-relay.service.ts

```diff
@@ -0,0 +1,349 @@
+import { Redactable } from '@/decorators/Redactable';
+import { EventRelay } from './event-relay.service';
+import type { AuditEventArgs } from './audit.types';
+import type { IWorkflowBase } from 'n8n-workflow';
+
+@Service()
+export class AuditEventRelay {
+	constructor(
+		private readonly eventRelay: EventRelay,
+		private readonly eventBus: MessageEventBus,
+	) {
+		this.setupHandlers();
+	}
+
+	private setupHandlers() {
+		this.eventRelay.on('workflow-created', (event) => this.workflowCreated(event));
+		this.eventRelay.on('workflow-deleted', (event) => this.workflowDeleted(event));
+		this.eventRelay.on('workflow-saved', (event) => this.workflowSaved(event));
+		this.eventRelay.on('workflow-pre-execute', (event) => this.workflowPreExecute(event));
+		this.eventRelay.on('workflow-post-execute', (event) => this.workflowPostExecute(event));
+		this.eventRelay.on('node-pre-execute', (event) => this.nodePreExecute(event));
+		this.eventRelay.on('node-post-execute', (event) => this.nodePostExecute(event));
+		this.eventRelay.on('user-deleted', (event) => this.userDeleted(event));
+		this.eventRelay.on('user-invited', (event) => this.userInvited(event));
+		this.eventRelay.on('user-reinvited', (event) => this.userReinvited(event));
+		this.eventRelay.on('user-updated', (event) => this.userUpdated(event));
+		this.eventRelay.on('user-signed-up', (event) => this.userSignedUp(event));
+		this.eventRelay.on('user-logged-in', (event) => this.userLoggedIn(event));
+		this.eventRelay.on('user-login-failed', (event) => this.userLoginFailed(event));
+		this.eventRelay.on('user-invite-email-click', (event) => this.userInviteEmailClick(event));
+		this.eventRelay.on('user-password-reset-email-click', (event) =>
+			this.userPasswordResetEmailClick(event),
+		);
+		this.eventRelay.on('user-password-reset-request-click', (event) =>
+			this.userPasswordResetRequestClick(event),
+		);
+		this.eventRelay.on('api-key-created', (event) => this.apiKeyCreated(event));
+		this.eventRelay.on('api-key-deleted', (event) => this.apiKeyDeleted(event));
+		this.eventRelay.on('email-failed', (event) => this.emailFailed(event));
+		this.eventRelay.on('credentials-created', (event) => this.credentialsCreated(event));
+		this.eventRelay.on('credentials-deleted', (event) => this.credentialsDeleted(event));
+		this.eventRelay.on('credentials-shared', (event) => this.credentialsShared(event));
+		this.eventRelay.on('credentials-updated', (event) => this.credentialsUpdated(event));
+		this.eventRelay.on('credentials-deleted', (event) => this.credentialsDeleted(event));
+		this.eventRelay.on('community-package-installed', (event) =>
+			this.communityPackageInstalled(event),
+		);
+		this.eventRelay.on('community-package-updated', (event) => this.communityPackageUpdated(event));
+		this.eventRelay.on('community-package-deleted', (event) => this.communityPackageDeleted(event));
+	}
+
+	/**
+	 * Workflow
+	 */
+
+	@Redactable()
```

Comment:
Wondering if it would be more future proof to do this other way around. I.e. anonymize everything unless stated otherwise. That way you can't forget to add the anonymization when adding a new event. Tho OK to leave that for future.

Reply:
https://github.com/n8n-io/n8n/pull/9724#discussion_r1642909544